$NetBSD: patch-aa,v 1.1 2009/07/24 15:18:24 tnn Exp $

add a local copy of SSL_SESSION_cmp which is missing in openssl 1.0.0 betas.
based on hack found at: http://trac.macports.org/ticket/19124

--- src/ne_openssl.c.orig	2008-10-30 21:23:54.000000000 +0100
+++ src/ne_openssl.c
@@ -593,6 +593,15 @@ void ne_ssl_context_destroy(ne_ssl_conte
     ne_free(ctx);
 }
 
+static int static_SSL_SESSION_cmp(const SSL_SESSION *a,const SSL_SESSION *b)
+{
+    if (a->ssl_version != b->ssl_version)
+        return(1);
+    if (a->session_id_length != b->session_id_length)
+        return(1);
+    return(memcmp(a->session_id, b->session_id, a->session_id_length));
+}
+
 /* For internal use only. */
 int ne__negotiate_ssl(ne_session *sess)
 {
@@ -675,7 +684,7 @@ int ne__negotiate_ssl(ne_session *sess)
     if (ctx->sess) {
         SSL_SESSION *newsess = SSL_get0_session(ssl);
         /* Replace the session if it has changed. */ 
-        if (newsess != ctx->sess || SSL_SESSION_cmp(ctx->sess, newsess)) {
+        if (newsess != ctx->sess || static_SSL_SESSION_cmp(ctx->sess, newsess)) {
             SSL_SESSION_free(ctx->sess);
             ctx->sess = SSL_get1_session(ssl); /* bumping the refcount */
         }
